src/Aspire.Hosting.Azure.Provisioning/Provisioners/AISearchProvisioner.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure;
+using Azure.ResourceManager.Search;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.Azure.Provisioning.Provisioners;
+
+internal sealed class AISearchProvisioner(ILogger<AISearchProvisioner> logger) : AzureResourceProvisioner<AzureAISearchResource>
+{
+    public override bool ConfigureResource(IConfiguration configuration, AzureAISearchResource resource)
+    {
+        if (configuration.GetConnectionString(resource.Name) is string connectionString)
+        {
+            resource.ConnectionString = connectionString;
+            return true;
+        }
+
+        return false;
+    }
+
+    public override async Task GetOrCreateResourceAsync(
+        AzureAISearchResource resource,
+        ProvisioningContext context,
+        CancellationToken cancellationToken)
+    {
+        context.ResourceMap.TryGetValue(resource.Name, out var azureResource);
+
+        if (azureResource is not null && azureResource is not SearchServiceResource)
+        {
+            logger.LogWarning("Resource {resourceName} is not an AI Search resource. Deleting it.", resource.Name);
+
+            await context.ArmClient.GetGenericResource(azureResource.Id).DeleteAsync(WaitUntil.Started, cancellationToken).ConfigureAwait(false);
+        }
+
+        var aiSearchResource = azureResource as SearchServiceResource;
+
+        if (aiSearchResource is null)
+        {
+            var aiSearchName = Guid.NewGuid().ToString().Replace("-", string.Empty);
+
+            var parameters = new SearchServiceData(context.Location);
```

Comment:
It'd be nice to allow folks to use the [Free Tier](https://learn.microsoft.com/en-us/azure/search/search-sku-tier) for getting started or local development.  It's worth thinking about whether we want to punt for now and get this for free with a better declarative resource provisioning story, or if we'd want to special case like we do for some of the Emulators in the hosting layer.

Reply:
Yes this should be optimized for dev/test (so free) by default.